docs/configuration/secure.md

```diff
@@ -0,0 +1,571 @@
+```
+
+see also: [Browser compatibility of HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP#Browser_compatibility)
+
+##### Apache
+Add the following line INSIDE the `` <VirtualHost></VirtualHost> `` block:
+
+```
+    <IfModule mod_headers.c>
+        Header set X-Content-Type-Options "nosniff"
+        Header set X-XSS-Protection "1; mode=block"
+        Header set X-Frame-Options "SAMEORIGIN;"
+        Header set Referrer-Policy "strict-origin"
+        Header set Feature-Policy "sync-xhr 'self'"
+
+        # Working with ILIAS in most use cases:
+        Header set Content-Security-Policy "default-src 'self'; connect-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' blob:; style-src 'self' 'unsafe-inline' data:; img-src 'self' 'unsafe-inline' data:; font-src 'self' 'unsafe-inline' data:; media-src 'self' 'unsafe-inline' data:"
+
+        # Working with ILIAS in most use cases:
+        Header set X-Content-Security-Policy "default-src 'self'; connect-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' blob:; style-src 'self' 'unsafe-inline' data:; img-src 'self' 'unsafe-inline' data:; font-src 'self' 'unsafe-inline' data:; media-src 'self' 'unsafe-inline' data:"
+    </IfModule>
+```
+
+ ##### NGINX
+ You can simply add this in your ```server``` configuration:
+
+ ```
+  # Content-Security
+         add_header X-Frame-Options "SAMEORIGIN";
+         add_header X-Content-Type-Options nosniff;
+         add_header X-XSS-Protection "1; mode=block";
+         add_header Referrer-Policy "strict-origin";
+         add_header Feature-Policy "sync-xhr 'self';
+
+         # Working with ILIAS in most use cases:
+         add_header Content-Security-Policy "default-src 'self'; connect-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' blob:; style-src 'self' 'unsafe-inline' data:; img-src 'self' 'unsafe-inline' data:; font-src 'self' 'unsafe-inline' data:; media-src 'self' 'unsafe-inline' data:";
+         #compatibility to  Microsoft Internet Explorer 10:
+         add_header X-Content-Security-Policy "default-src 'self'; connect-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' blob:; style-src 'self' 'unsafe-inline' data:; img-src 'self' 'unsafe-inline' data:; font-src 'self' 'unsafe-inline' data:; media-src 'self' 'unsafe-inline' data:";
+ ```
+
+note:  
+If you use a proxied [Chat Server](https://github.com/ILIAS-eLearning/ILIAS/blob/release_5-4/Modules/Chatroom/README.md), you MUST add the url to the CSP definition:  
+```connect-src 'self' wss://onscreenchat.yourservername.org https://onscreenchat.yourservername.org;```
+
+#### Validate the header configuration
+It is recommended to validate your configuration with the services from https://securityheaders.com/.  
+Try to reach **A** grade.
+
+### Enable OCSP stapling (TLS Certificate Status Request)
```

Comment:
OCSP stapling is more a privacy than a security feature. Maybe we should just link to some external documentation here as the prerequirements for OCSP stapling may vary a lot (some environments might force you to download your own OCSP files, don't have access to DNS, etc.).

Reply:
It will also improve the performance and also add more privacy since certificates are checked for actual validity or if they are revoked/compromised.
So we can provide this detailed and not only less by a reference.